pyuvdata/calfits.py

```diff
@@ -0,0 +1,194 @@
+        prihdr['NANTSTEL'] = self.Nants_telescope
+        prihdr['NSPWS'] = self.Nspws
+        prihdr['XORIENT'] = self.x_orientation
+        for line in self.history.splitlines():
+            prihdr.add_history(line)
+
+        for p in self.extra():
+            ep = getattr(self, p)
+            if ep.form is 'str':
+                prihdr['{0}'.format(p.upper().replace('_', '')[:8])] = ep.value
+            else:
+                continue
+
+        # if self.pipeline: prihdr['CALPIPE'] = self.pipeline
+        # if self.observer: prihdr['OBSERVER'] = self.observer
+        # if self.git_origin: prihdr['ORIGIN'] = self.git_origin
+        # if self.git_hash: prihdr['HASH'] = self.git_hash
+
+        if self.cal_type == 'gain':
+            # Set header variable for gain.
+            prihdr['CTYPE4'] = ('FREQS', 'Frequency.')
+            prihdr['CUNIT4'] = ('Hz', 'Units of frequecy.')
+            prihdr['CRVAL4'] = self.freq_array[0][0]
+            prihdr['CDELT4'] = self.channel_width
+
+            # set the last axis for number of arrays.
+            prihdr['CTYPE1'] = ('Narrays', 'Number of image arrays.')
+            prihdr['CUNIT1'] = ('Integer', 'Number of image arrays. Increment.')
+            prihdr['CRVAL1'] = (3, 'Number of image arryays.')
+            prihdr['CDELT1'] = 1
+
+            pridata = np.concatenate([self.gain_array.real[:, :, :, :, np.newaxis],
+                                      self.gain_array.imag[:, :, :, :, np.newaxis],
+                                      # self.flag_array[:, :, :, :, np.newaxis],
+                                      self.quality_array[:, :, :, :, np.newaxis]],
+                                     axis=-1)
+
+        if self.cal_type == 'delay':
+            # Set header variable for gain.
+            prihdr['CTYPE4'] = ('FREQS', 'Valid frequencies to apply delay.')
+            prihdr['CUNIT4'] = ('Hz', 'Units of frequecy.')
+            prihdr['CRVAL4'] = self.freq_array[0][0]
+            prihdr['CDELT4'] = self.channel_width
+            # set the last axis for number of arrays.
+            prihdr['CTYPE1'] = ('Narrays', 'Number of image arrays.')
+            prihdr['CUNIT1'] = ('Integer', 'Number of image arrays. Value.')
+            prihdr['CRVAL1'] = (2, 'Number of image arrays.')
+            prihdr['CDELT1'] = 1
+
+            pridata = np.concatenate([self.delay_array[:, :, :, :, np.newaxis],
+                                      # self.flag_array[:, :, :, :, np.newaxis],
+                                      self.quality_array[:, :, :, :, np.newaxis]],
+                                     axis=-1)
+
+        if self.cal_type == 'unknown':
+            raise ValueError("unknown calibration type. Do not know how to"
+                             "store parameters")
+
+        # header ctypes for NAXIS
+        prihdr['CTYPE5'] = ('ANTS', 'Antenna numbering.')
+        prihdr['CUNIT5'] = 'Integer'
+        prihdr['CRVAL5'] = 0
+        prihdr['CDELT5'] = -1
```

Comment:
I'm a little worried that you're starting this array at zero and then incrementing by -1, are you sure this is what you want? Also, I'm not sure what 'Antenna numbering' in your comment means because you also have a variable called antenna_numbers that I don't think is related to this.